testutil/network/network.go

```diff
@@ -165,15 +166,87 @@ type (
 	}
 )
 
+type TestnetEnv interface {
```

Comment:
I don't love this name, but couldn't think of anything better

Reply:
I don't love this interface, it's imo mixing a bunch of stuff. I think we can go with only `Log` & `Logf`, so that it's more like a logger interface. (see below)